driver/ppm_events_public.h

```diff
@@ -418,6 +418,12 @@ along with sysdig.  If not, see <http://www.gnu.org/licenses/>.
 #define PPM_IPC_NOWAIT		(1 << 0)
 #define PPM_SEM_UNDO		(1 << 1)
 
+/*
+ * Semget flags
+ */
+#define PPM_IPC_CREAT  (1 << 13)
+#define PPM_IPC_EXCL   (1 << 14)
+
```

Comment:
It is great that you brought this one up @luca3m!

Let me describe the "problem" anyway (short story, happy to start from zero if it feels better). That integer parameter in semget encodes those two flags.. and ONLY these two flags are semaphore specific.

However, in the lower bits, that integer ALSO encodes a file mode (i.e. 0666, 0644, 0755, etc..). Posix defines a logical name for each of the bits on the file mode, they are:

```
S_IREAD
S_IWRITE
S_IEXEC
S_IRGRP
S_IWGRP
S_IXGRP
S_IROTH
S_IWOTH
S_IXOTH
S_ISUID
S_ISGID
S_ISVTX
```

Of course, the mode bits are present in LOTS of system calls, not only semaphores.. even the plain old `open` call has a mode integer with the same bit meaning as these ones.

Looking into sysdig I realized that mode bit decoding was unimplemented... so I left it as is.

After all that long story, that is where the 13 offset came from, I thought I would leave the earlier bits to encode mode info so that if someone else implements that in the future we don't break trace compatibility.

All that being said, I agree that perhaps the cost of making the offsets seem a bit magical/absurd is not worth it. What do you think?